packages/core/src/components/forms/_input-group.scss

```diff
@@ -233,6 +233,10 @@ $input-button-height-small: $pt-button-height-smaller !default;
     &.#{$ns}-intent-#{$intent} {
       .#{$ns}-input {
         @include pt-input-intent($color);
+
+        .#{$ns}-dark & {
+          @include pt-dark-input-intent(map-get($pt-dark-input-intent-box-shadow-colors, $intent));
```

Comment:
use of `$color4` for dark mode intent input within input group moved to here
![Screen Shot 2021-12-01 at 16 41 41](https://user-images.githubusercontent.com/5934779/144318594-67a60457-8097-459a-9feb-85fdcb770c50.png)